tests/CordovaLibTests/CDVWhitelistTests.m

```diff
@@ -295,6 +295,37 @@ - (void)testAllowIntentsAndNavigations
     XCTAssertEqual([CDVIntentAndNavigationFilter filterUrl:[NSURL URLWithString:@"https://google.com"] intentsWhitelist:intentsWhitelist navigationsWhitelist:navigationsWhitelist], CDVIntentAndNavigationFilterValueIntentAllowed);
     // Test http (not allowed in either)
     XCTAssertEqual([CDVIntentAndNavigationFilter filterUrl:[NSURL URLWithString:@"http://google.com"] intentsWhitelist:intentsWhitelist navigationsWhitelist:navigationsWhitelist], CDVIntentAndNavigationFilterValueNoneAllowed);
+    
+    
+    NSURL* telUrl = [NSURL URLWithString:@"tel:5555555"];
+    NSMutableURLRequest* telRequest = [NSMutableURLRequest requestWithURL:telUrl];
+    telRequest.mainDocumentURL = telUrl;
+    
+    // mainDocumentURL and URL are the same in the NSURLRequest
+    // Only UIWebViewNavigationTypeLinkClicked and UIWebViewNavigationTypeOther should return YES
+    XCTAssertTrue([CDVIntentAndNavigationFilter shouldOpenURLRequest:telRequest navigationType:UIWebViewNavigationTypeLinkClicked]);
+    XCTAssertTrue([CDVIntentAndNavigationFilter shouldOpenURLRequest:telRequest navigationType:UIWebViewNavigationTypeOther]);
+    XCTAssertFalse([CDVIntentAndNavigationFilter shouldOpenURLRequest:telRequest navigationType:UIWebViewNavigationTypeReload]);
+    XCTAssertFalse([CDVIntentAndNavigationFilter shouldOpenURLRequest:telRequest navigationType:UIWebViewNavigationTypeBackForward]);
+    XCTAssertFalse([CDVIntentAndNavigationFilter shouldOpenURLRequest:telRequest navigationType:UIWebViewNavigationTypeFormSubmitted]);
+    XCTAssertFalse([CDVIntentAndNavigationFilter shouldOpenURLRequest:telRequest navigationType:UIWebViewNavigationTypeFormResubmitted]);
+    
```

Comment:
See my comment on that note. The operative test here is for "shouldOverrideLoadWithRequest" -- if the intent is allowed, it should return NO, as in it shouldn't override the load with the request (since it is opened externally).